test/FunctionalTests/Microsoft.OData.Client.Tests/ALinq/DollarApplyGroupByTests.cs

```diff
@@ -0,0 +1,1717 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="DollarApplyGroupByTests.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.OData.Client.Tests.ALinq
+{
+    public class DollarApplyGroupByTests : DollarApplyTestsBase
+    {
+        public DollarApplyGroupByTests() : base()
+        {
+        }
+
+        [Fact]
+        public void GroupByResultSelector_Sum_ByConstant()
+        {
+            var queryable = this.dsContext.CreateQuery<Number>(numbersEntitySetName);
+
+            var aggregateQuery = queryable.GroupBy(d1 => 1, (d2, d3) => new
+            {
+                SumIntProp = d3.Sum(d4 => d4.IntProp),
+                SumNullableIntProp = d3.Sum(d4 => d4.NullableIntProp),
+                SumDoubleProp = d3.Sum(d4 => d4.DoubleProp),
+                SumNullableDoubleProp = d3.Sum(d4 => d4.NullableDoubleProp),
+                SumDecimalProp = d3.Sum(d4 => d4.DecimalProp),
+                SumNullableDecimalProp = d3.Sum(d4 => d4.NullableDecimalProp),
+                SumLongProp = d3.Sum(d4 => d4.LongProp),
+                SumNullableLongProp = d3.Sum(d4 => d4.NullableLongProp),
+                SumSingleProp = d3.Sum(d4 => d4.SingleProp),
+                SumNullableSingleProp = d3.Sum(d4 => d4.NullableSingleProp),
+            });
+
+            Assert.Equal(
+                string.Format(
+                    "{0}/Numbers?$apply=aggregate(" +
+                    "IntProp with sum as SumIntProp,NullableIntProp with sum as SumNullableIntProp," +

```

Comment:
Can a single property be referenced multiple times in the aggregation results? For example if you want to get both the sum and average of `IntProp`?

Reply:
Yes. For example, `Amount` is repeated multiple times in `$apply=aggregate(Amount with average as AverageAmount,Amount with sum as SumAmount)`